app/_how-tos/configure-oidc-with-consumers.md

```diff
@@ -0,0 +1,166 @@
+      Yes, you can combine the Consumer claim with the authorization group claim to further secure your environment. See the how-to on [configuring OIDC with ACL groups](/how-to/configure-oidc-with-acl-auth/) for more information.
+
+cleanup:
+  inline:
+    - title: Clean up Konnect environment
+      include_content: cleanup/platform/konnect
+      icon_url: /assets/icons/gateway.svg
+    - title: Destroy the {{site.base_gateway}} container
+      include_content: cleanup/products/gateway
+      icon_url: /assets/icons/gateway.svg
+
+---
+
+## 1. Enable the OpenID Connect plugin
+
+Using the Keycloak and {{site.base_gateway}} configuration from the [prerequisites](#prerequisites), 
+set up an instance of the OpenID Connect plugin. In this example, we're using the simple password grant with the `preferred_username` Consumer claim.
+
+Enable the OpenID Connect plugin on the `example-service` Service:
+
+{% entity_examples %}
+entities:
+  plugins:
+    - name: openid-connect
+      service: example-service
+      config:
+        issuer: ${issuer}
+        client_id:
+        - ${client-id}
+        client_secret:
+        - ${client-secret}
+        client_auth:
+        - client_secret_post
+        auth_methods:
+        - password
+        consumer_claim:
+        - preferred_username
+        consumer_by:
+        - username
+variables:
+  issuer:
+    value: $ISSUER
+  client-id:
+    value: $CLIENT_ID
+  client-secret:
+    value: $CLIENT_SECRET
+{% endentity_examples %}
+
+In this example:
+* `issuer`, `client ID`, `client secret`, and `client auth`: Settings that connect the plugin to your IdP (in this case, the sample Keycloak app).
+* `auth_methods`: Password grant, for easy testing.
+* `consumer_claim` and `consumer_by` : Looks for a `preferred_username` in the token payload and maps it to the Consumer entity by the entity's `username` value.
+
+{% include_cached plugins/oidc/client-auth.md %}
+
+## 2. Create a Consumer
+
+First, let's try to access the Service without a matching Consumer.
+Request the Service with the basic authentication credentials created in the [prerequisites](#prerequisites):
+
+{% validation request-check %}
+url: /anything
+method: GET
+status_code: 403
+user: "john:doe"
+display_headers: true
+{% endvalidation %}
+
+You should get a `403 Forbidden` error code, which means the Service is protected by authentication.
+
+Create a Consumer with a username that matches the user in your IdP, in this case `john`:
+
+{% entity_examples %}
+entities:
+  consumers:
+    - username: john
+{% endentity_examples %}
+
+## 3. Verify Consumer authorization
+
+At this point you have created a Gateway Service, routed traffic to the Service, enabled the OpenID Connect plugin, and created a Consumer that maps to a `preferred_username` of an existing user.
+
+Now, your configured Consumer can access the `example-route` Route:
+
+{% validation request-check %}
+url: /anything
+method: GET
+status_code: 403
+user: "john:doe"
```

Comment:
Can we also specify the password to set in the keycloak prereq, like `password` so that users can copy/paste this directly?